specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json

```diff
@@ -0,0 +1,2533 @@
+            "identity": {
+              "$ref": "#/definitions/ManagedClusterIdentity",
+              "description": "The identity of the managed cluster, if configured."
+            }
+          }
+        }
+      ],
+      "description": "Managed cluster."
+    },
+    "ManagedClusterProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The current deployment or provisioning state, which only appears in the response."
+        },
+        "maxAgentPools": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "The max number of agent pools for the managed cluster."
+        },
+        "kubernetesVersion": {
+          "type": "string",
+          "description": "Version of Kubernetes specified when creating the managed cluster."
+        },
+        "dnsPrefix": {
+          "type": "string",
+          "description": "DNS prefix specified when creating the managed cluster."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "FQDN for the master pool."
+        },
+        "privateFQDN": {
+          "readOnly": true,
+          "type": "string",
+          "description": "FQDN of private cluster."
+        },
+        "agentPoolProfiles": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterAgentPoolProfile"
+          },
+          "description": "Properties of the agent pool."
+        },
+        "linuxProfile": {
+          "$ref": "#/definitions/ContainerServiceLinuxProfile",
+          "description": "Profile for Linux VMs in the container service cluster."
+        },
+        "windowsProfile": {
+          "$ref": "#/definitions/ManagedClusterWindowsProfile",
+          "description": "Profile for Windows VMs in the container service cluster."
+        },
+        "servicePrincipalProfile": {
+          "$ref": "#/definitions/ManagedClusterServicePrincipalProfile",
+          "description": "Information about a service principal identity for the cluster to use for manipulating Azure APIs."
+        },
+        "addonProfiles": {
+          "additionalProperties": {
+            "$ref": "#/definitions/ManagedClusterAddonProfile"
+          },
+          "description": "Profile of managed cluster add-on."
+        },
+        "nodeResourceGroup": {
+          "type": "string",
+          "description": "Name of the resource group containing agent pool nodes."
+        },
+        "enableRBAC": {
+          "type": "boolean",
+          "description": "Whether to enable Kubernetes Role-Based Access Control."
+        },
+        "enablePodSecurityPolicy": {
+          "type": "boolean",
+          "description": "(PREVIEW) Whether to enable Kubernetes Pod security policy."
+        },
+        "networkProfile": {
+          "$ref": "#/definitions/
```

Comment:
>scan-interval [](start = 13, length = 13)

I discussed this on another PR. Generally the naming convention is lowerCaseCamelCase for property names, but since these are passed through to a Kubernetes extension, I think there's value in keeping them as-is. Otherwise, the users will have to create a mental mapping between `scanInterval` becoming `scan-interval` and some for all others.

Reply:
Yup, for instance the Python SDK would show it as "scanInterval" in the GET MC response. But for serializing and deserializing, they'd do the conversion to/from "scan-interval" properly. I think it's a tradeoff here to achieve recognizable parameters.